internal/common/server_status.go

```diff
@@ -82,34 +92,33 @@ func WriteAndHandleStatusFile() error {
 	return nil
 }
 
-// Returns (true, nil) if it's safe to create the status file,
-// (false, nil) if another instance of the server is running,
-// or (false, err) if there are issues reading or removing the status file.
-func checkAndHandleStalePid() (bool, error) {
+// Returns (nil, nil) if the status file doesn't exist,
+// (&statusJson, nil) if the status file exists and another instance of the server is running,
+// or (nil, err) if there are issues reading or removing the status file.
```

Comment:
You can streamline this a bit by just saying something like:
"When there is no error, ...".
This just assumes (fairly) that there will always be a nil returned for the first value on an error.